### ISSUE TITLE
Return rejected promises

### ISSUE THREAD

**Issue by Sicria** [nytimes/react-tracking#106]
After updating from ``5.3.0`` > ``5.6.0`` the decorator now does not return the value of a promise.

```
@track(props => ({
  action: 'FormSubmit',
  label: props.name || null,
}))
handleSubmit = values => this.props.onSubmit(values).then(() => {}, err => err);
```

The ``onSubmit`` function above returns a promise which returns an error to display submission errors via ``react-final-form``. If the decorator is used this error does not get reported as the value isn't passed through.

**Comment by tizmagik** [comment]
Thanks for submitting an issue. My guess is this behavior changed in [v5.4.0](https://github.com/NYTimes/react-tracking/releases/tag/v5.4.0).

For your use case I'm not sure I understand what is the desired behavior and whether the [new async/promise support mentioned in the docs](https://github.com/NYTimes/react-tracking#advanced-usage) would help your particular use case?

**Comment by tizmagik** [comment]
Fixed in #147 and released as [v7.3.0](https://github.com/nytimes/react-tracking/releases/tag/v7.3.0) -- thanks for reporting! 🎉
